solver/src/settlement_submission/archer_settlement.rs

```diff
@@ -0,0 +1,377 @@
+                .append_to_execution_plan(block_coinbase::PayBlockCoinbase {
+                    amount: tx_gas_cost_in_ether_wei,
+                });
+            let method = super::retry::settle_method_builder(&self.contract, settlement.into())
+                .from(self.account.clone())
+                .nonce(nonce)
+                // Wouldn't work because the function isn't payable.
+                // .value(tx_gas_cost_in_ether_wei)
+                .gas(U256::from_f64_lossy(gas_limit))
+                .gas_price(GasPrice::Value(U256::zero()));
+
+            // simulate transaction
+
+            if let Err(err) = method.clone().view().call().await {
+                if let Some((_, previous_tx)) = previous_tx.as_ref() {
+                    if let Err(err) = self.archer_api.cancel(previous_tx).await {
+                        tracing::error!("archer cancellation failed: {:?}", err);
+                    }
+                }
+                return err;
+            }
+
+            // If gas price has increased cancel old and submit new new transaction.
+
+            if let Some((previous_gas_price, previous_tx)) = previous_tx.as_ref() {
+                if gas_price > *previous_gas_price {
+                    if let Err(err) = self.archer_api.cancel(previous_tx).await {
+                        tracing::error!("archer cancellation failed: {:?}", err);
+                    }
+                } else {
+                    tokio::time::sleep(UPDATE_INTERVAL).await;
+                    continue;
+                }
+            }
+
+            // Unwrap because no communication with the node is needed because we specified nonce and gas.
+            let (raw_signed_transaction, hash) =
+                match method.tx.build().now_or_never().unwrap().unwrap() {
+                    Transaction::Request(_) => unreachable!("used local account"),
+                    Transaction::Raw { bytes, hash } => (bytes.0, hash),
+                };
+
+            tracing::info!(
+                "creating archer transaction with hash {:?}, tip to miner {:.3e}, gas price {:.3e}, gas estimate {}",
+                hash,
+                tx_gas_cost_in_ether_wei.to_f64_lossy(),
+                gas_price,
+                gas_estimate,
+            );
+
+            if let Err(err) = self
+                .archer_api
+                .submit_transaction(&raw_signed_transaction, deadline)
+                .await
+            {
+                tracing::error!("archer submission failed: {:?}", err);
+                tokio::time::sleep(UPDATE_INTERVAL).await;
+                continue;
+            }
+
+            transactions.push(hash);
+            previous_tx = Some((gas_price, raw_signed_transaction));
+            tokio::time::sleep(UPDATE_INTERVAL).await;
+        }
+    }
+}
+
+/// From a list of potential hashes find one that was mined.
+async fn find_mined_transaction(web3: &Web3, hashes: &[H256]) -> Option<H256> {
```

Comment:
What we could do is an `eth_getLogs` for `Settlement` log with a topic filter for the solver address you are looking for. We should be able to do this with a single ETHRPC request.

I'm 100% fine with this solution as well.